src/python/pants/engine/internals/graph.py

```diff
@@ -168,85 +175,170 @@ def warn_deprecated_field_type(request: _WarnDeprecatedFieldRequest) -> _WarnDep
+    addressable using its un-parameterized Address.
+    """
+
+    target: Target | None
+    parametrizations: FrozenDict[Address, Target]
+
+    def get(self, address: Address) -> Target | None:
+        if self.target and self.target.address == address:
+            return self.target
+        return self.parametrizations.get(address)
+
+    def generated_or_generator(self, maybe_generator: Address) -> Iterator[Target]:
+        if not self.target:
+            raise ValueError(
+                "A `parametrized` target cannot be consumed without its parameters specified.\n"
+                f"Target `{maybe_generator}` can be addressed as:\n"
+                f"{bullet_list(addr.spec for addr in self.parametrizations)}"
+            )
+        if self.parametrizations:
+            # Generated Targets.
+            yield from self.parametrizations.values()
+        else:
+            # Did not generate targets.
+            yield self.target
+
+
 @rule
-async def resolve_target(
+async def resolve_target_parametrizations(
     address: Address,
     registered_target_types: RegisteredTargetTypes,
     union_membership: UnionMembership,
     target_types_to_generate_requests: TargetTypesToGenerateTargetsRequests,
-) -> WrappedTarget:
-    if not address.is_generated_target:
-        target_adaptor = await Get(TargetAdaptor, Address, address)
-        target_type = registered_target_types.aliases_to_types.get(target_adaptor.type_alias, None)
-        if target_type is None:
-            raise UnrecognizedTargetTypeException(
-                target_adaptor.type_alias, registered_target_types, address
+    files_not_found_behavior: FilesNotFoundBehavior,
+) -> _TargetParametrizations:
+    assert not address.is_generated_target and not address.is_parametrized
+
+    target_adaptor = await Get(TargetAdaptor, Address, address)
+    target_type = registered_target_types.aliases_to_types.get(target_adaptor.type_alias, None)
+    if target_type is None:
+        raise UnrecognizedTargetTypeException(
+            target_adaptor.type_alias, registered_target_types, address
+        )
+    if (
+        target_type.deprecated_alias is not None
+        and target_type.deprecated_alias == target_adaptor.type_alias
+        and not address.is_generated_target
+    ):
+        await Get(_WarnDeprecatedTarget, _WarnDeprecatedTargetRequest(target_type))
+
+    target: Target
+    generate_request = target_types_to_generate_requests.request_for(target_type)
+    if generate_request:
+        # Split out the `propagated_fields` before construction.
+        generator_fields = dict(target_adaptor.kwargs)
+        template_fields = {}
+        # TODO: Require for all instances before landing.
+        if issubclass(target_type, TargetGenerator):
+            copied_fields = (
+                *target_type.copied_fields,
+                *target_type._find_plugin_fields(union_membership),
```

Comment:
Ah ha! This was a major question I had. It's fine in a followup, but it's important that we have a dedicated test for this, I think, given that we now have an allowlist of copied fields whereas we used to just copy _everything_.

Reply:
Yea, I'd appreciate your thoughts on what to do with plugin fields, although I don't think that it blocks.

Reply:
Every instance we have right now is indeed a Copy, and by definition, it can't have fancier semantics because the generation rule will not know about it in advance.